src/monster.cpp

```diff
@@ -362,15 +362,37 @@ void monster::try_reproduce()
 
 void monster::refill_udders()
 {
-    if( !has_flag( MF_MILKABLE ) ) {
-        return;
+    auto milked_item = type->starting_ammo.find( "milk_raw " );
+    if( milked_item == type->starting_ammo.end() ) {
+        milked_item = type->starting_ammo.find( "milk" );
+        if( milked_item == type->starting_ammo.end() ) {
+            // this may be a sheep from a very old save.
+            // sheep didnt use to be milkable, but now they have that flag,
+            // but the old save would keep their old blank starting ammo.
+            // cant modify the type, and cant get what their starting ammo should be.
```

Comment:
Why can't you get what their starting ammo is? What exactly does `type->starting_ammo` contain, if not the starting ammo?

`MonsterGenerator::check_monster_definitions` should check for this problem (a monster type with milkable flag, but no milk starting ammo) and issue a debug message. This problem should not appear out of nowhere in the middle of the game.

Reply:
Its currently the case on the attached save in the linked issue, 

The sheep in that save has neither milk nor milk_raw in their starting ammo, and yet they have the milkable flag.

Reply:
> The sheep in that save has neither milk nor milk_raw in their starting ammo, and yet they have the milkable flag.

Well, *that* is the problem. The game should not allow this case as it can't handle this. This should be checked for when the game data is loaded, not in the middle of the game when the player stumbles upon that case.

Reply:
Issuing a debugmsg would be ok on that mismatch on load? I dont know whether that sheep in that save in the linked issue, is a weird case, or just what happens when very very old savegames are loaded from before sheep were milkable.

( it gives them the milkable flag, so I dont kno wwhy it didnt give them milk? ) if the sheep in that save is just a very weird edge case, then I guess I can do the debugmsg on load if theres a mismatch

Reply:
Here is a guess (or at least another bug): when a lamb matures (via `monster::try_upgrade`), it becomes a sheep by calling `monster::poly`. That function changes the type of the monster, but does not change its current starting ammo. So the adult sheep keeps the starting ammo of the lamb (which is unsurprising not containing milk). That could be the cause of the bug in the linked game.

Or it is just another bug.

Reply:
Nevermind, Im an absolute idiot, the sheep type starting_ammo is valid, I got my checks mixed up, knew it was weird that was happening.

Reply:
Ah, that is something else to consider yeah, my latest commit fixes the upgrade from milk to raw_milk, but dosnt hardcode the need for milk ( so mods can do weird stuff and define chocolate_milk or something if they want )

I think the latest commit should fix this case, as, when it changes to mon_sheep from lamb, its type still has the sheeps starting_ammo defined, and if its milkable, yet has empty ammo but type->starting ammo defined, the function will fill it up with its type->starting_ammo

`    if( type->starting_ammo.empty() ) {
        debugmsg( "monster %s has no starting ammo to refill udders", get_name() );
        return;
    }
    if( ammo.empty() ) {
        // legacy animals got empty ammo map, fill them up now if needed.
        ammo[type->starting_ammo.begin()->first] = type->starting_ammo.begin()->second;
    }`

If the type->starting_ammo is empt,y thats the real problem, but luckily, thats not actually happening, i was wrongf before.